rosbag2_storage/include/rosbag2_storage/storage_interfaces/base_read_interface.hpp

```diff
@@ -25,6 +25,15 @@
 
 namespace rosbag2_storage
 {
+
+enum class ReadOrder
+{
+  // Read messages in increasing timestamp order
+  TIMESTAMP,
+  // Read messages in decreasing timestamp order
+  REVERSE_TIMESTAMP
```

Comment:
may also want to specify "file" or "physical" order, meaning the order the messages are present in the file. This might be good for debugging a bag with incorrect timestamps.

Reply:
i was wondering at first if that made sense for SQL, but there is a `messages.id` that I can order the query on separately from timestamp, which is the most logical "file order" there, so I'll do that

Reply:
Done, added file order